database/migrations/2023_07_04_010134_create_schema.php

```diff
@@ -0,0 +1,179 @@
+            $table->id();
+            $table->foreignIdFor(\App\Models\User::class)->unique()->constrained()->cascadeOnDelete();
+            $table->foreignIdFor(\App\Models\SocialMedia::class)->nullable()->constrained()->cascadeOnDelete();
+            $table->timestamps();
+        });
+        Schema::create('companies', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(\App\Models\User::class)->unique()->constrained()->cascadeOnDelete();
+            $table->foreignIdFor(\App\Models\SocialMedia::class)->nullable()->constrained()->cascadeOnDelete();
+            $table->timestamps();
+        });
+        Schema::create('admins', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(\App\Models\User::class)->unique()->constrained()->cascadeOnDelete();
+            $table->timestamps();
+        });
+
+        // SINF tables
+        Schema::create('editions', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('year');
+            $table->timestamps();
+        });
+        Schema::create('products', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->integer('price')->unsigned();
+            $table->integer('stock')->unsigned();
+            $table->foreignIdFor(\App\Models\Edition::class)->constrained()->cascadeOnDelete();
+            $table->timestamps();
+        });
+
+        // Activity tables
+        Schema::create('events', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->dateTime('date_start');
+            $table->dateTime('date_end');
+            $table->string('topic');
+            $table->integer('capacity')->unsigned()->nullable();
+            $table->foreignIdFor(\App\Models\Edition::class)->constrained()->cascadeOnDelete();
+            $table->timestamps();
+        });
+        Schema::create('speakers', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('title')->nullable();
+            $table->string('description')->nullable();
+            $table->string('organization')->nullable();
+            $table->foreignIdFor(\App\Models\SocialMedia::class)->nullable()->constrained()->cascadeOnDelete();
+            $table->foreignIdFor(\App\Models\Event::class)->constrained()->cascadeOnDelete();
+            $table->timestamps();
+        });
+
+        // Quest tables
+        Schema::create('quests', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->integer('points')->unsigned();
+
+            // we need to be careful to update this when we add new categories
+            $table->enum('category', ['COMPANY', 'TALK', 'WORKSHOP', 'MILESTONE', 'TEAMBUILDING']);
+
+            $table->morphs('redeemable');
```

Comment:
I'm probably nitpicking, but if you're referring to the reward attribute on the Quest table of the UML, shouldn't you also name it as it is in the diagram? Again, it's a really insignificant detail, but still, I think we should remain 100% faithful to the names we gave in the UML
```suggestion
            $table->morphs('reward');
```

Reply:
That's not the reward, that's the event/company that needs to be redeemed to unlock the quest, redeemable may not be the best name tho, maybe requirement?

Reply:
yes, it's a good replacement in my opinion. At least it's a more impartial and straightforward name